bindings/python/pyproject.toml

```
@@ -25,14 +25,13 @@ classifiers = [
   "Intended Audience :: Developers",
   "License :: OSI Approved :: Apache Software License",
   "Operating System :: OS Independent",
-  "Programming Language :: Python :: 3.9",
   "Programming Language :: Python :: 3.10",
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
 ]
 name = "pyiceberg-core"
 readme = "project-description.md"
-requires-python = "~=3.9"
+requires-python = ">=3.10,<3.13"
```

```suggestion
requires-python = ">=3.10"
```
nit: lets not add upperbound restrictions